project/NativeImage.scala

```diff
@@ -320,6 +320,43 @@ object NativeImage {
       }
     }
 
+  def checkNativeImageSize(
+    name: String,
+    targetDir: File
+  ): Def.Initialize[Task[Unit]] = Def.task {
+    val generatedBin = artifactFile(targetDir, name)
+    val logger       = streams.value.log
+    if (!generatedBin.exists) {
+      logger.error(s"Generated binary $generatedBin does not exist.")
+      logger.error(
+        "Ensure that the dependency on `buildNativeImage` is properly set."
+      )
+    }
+    val bytes              = generatedBin.attributes.size()
+    val productionNIBounds = (150, 450)
```

Comment:
Based on latest nightly build:
- Linux production NI has [433 MB](https://github.com/enso-org/enso/actions/runs/14827919978/job/41658717803#step:10:2148)
- Windows production NI has [390 MB](https://github.com/enso-org/enso/actions/runs/14827919978/job/41658719008#step:10:2207)
- MacOS aarch64 production NI has [393 MB](https://github.com/enso-org/enso/actions/runs/14827919978/job/41658718183#step:10:2482)
- MacOS x64 production NI has [378 MB](https://github.com/enso-org/enso/actions/runs/14827919978/job/41658718535#step:10:2310)

Reply:
Maybe the `maxImageSizeInMb` function should return different value for different OS to allow incremental individual improvements on each platform